tracer/src/Datadog.Trace.Tools.Analyzers/ConfigurationAnalyzers/EnvironmentGetEnvironmentVariableAnalyzer.cs

```diff
@@ -0,0 +1,255 @@
+            SyntaxNodeAnalysisContext context,
+            ArgumentSyntax argument)
+        {
+            var expression = argument.Expression;
+
+            // Only accept direct member access to ConfigurationKeys or PlatformKeys constants
+            // Example: ConfigurationKeys.ApiKey, ConfigurationKeys.CIVisibility.Enabled, PlatformKeys.SomeKey
+            if (expression is MemberAccessExpressionSyntax memberAccess)
+            {
+                if (IsValidConfigurationConstant(memberAccess, context.SemanticModel))
+                {
+                    // Valid: This is a ConfigurationKeys or PlatformKeys constant
+                    return;
+                }
+
+                // Invalid: Member access but not from ConfigurationKeys or PlatformKeys
+                var diagnostic = Diagnostic.Create(
+                    UseConfigurationKeysNotVariablesRule,
+                    memberAccess.GetLocation(),
+                    memberAccess.ToString());
+                context.ReportDiagnostic(diagnostic);
+                return;
+            }
+
+            // Everything else is invalid
+            string errorMessage;
+            DiagnosticDescriptor rule;
+
+            if (expression is LiteralExpressionSyntax literal && literal.Token.IsKind(SyntaxKind.StringLiteralToken))
+            {
+                // Hardcoded string literal
+                errorMessage = literal.Token.ValueText;
+                rule = UseConfigurationKeysRule;
+            }
+            else
+            {
+                // Variables, parameters, method calls, expressions, etc.
+                errorMessage = expression.ToString();
+                rule = UseConfigurationKeysNotVariablesRule;
+            }
+
+            var invalidDiagnostic = Diagnostic.Create(
+                rule,
+                expression.GetLocation(),
+                errorMessage);
+            context.ReportDiagnostic(invalidDiagnostic);
+        }
+
+        /// <summary>
+        /// Checks if a member access expression is a constant from ConfigurationKeys or PlatformKeys class.
+        /// Accepts: ConfigurationKeys.ApiKey, ConfigurationKeys.CIVisibility.Enabled, PlatformKeys.*, etc.
+        /// </summary>
+        private static bool IsValidConfigurationConstant(MemberAccessExpressionSyntax memberAccess, SemanticModel semanticModel)
+        {
+            var symbolInfo = semanticModel.GetSymbolInfo(memberAccess);
+            if (symbolInfo.Symbol is IFieldSymbol field)
+            {
+                // Check if it's a const field
+                if (!field.IsConst)
+                {
+                    return false;
+                }
+
+                // Check if it's from ConfigurationKeys, PlatformKeys, or any nested class within them
+                var containingType = field.ContainingType;
+                while (containingType != null)
+                {
+                    var fullTypeName = containingType.ToDisplayString();
```

Comment:
Again, this is kind of expensive, if we're using symbols, we should probably find the symbols ahead of time and use default symbol comparison instead